x-pack/test/functional/apps/lens/group2/fields_list.ts

```diff
@@ -263,16 +269,26 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
 
       it('should show new fields Available fields', async () => {
         await es.transport.request({
-          path: '/field-update-test/_doc',
-          method: 'POST',
+          path: '/field-update-test/_doc/1',
+          method: 'PUT',
           body: {
             '@timestamp': new Date().toISOString(),
-            oldField: 10,
+            oldField: 20,
             newField: 20,
           },
         });
+
+        await retry.try(async () => {
+          // wait for the new field to be available, at the first run it might be empty
+          const result = await es.transport.request({
+            path: '/field-update-test/_field_caps?fields=newField&include_empty_fields=false',
+            method: 'GET',
+          });
+          expect(result.fields.newField).to.be.ok();
+        });
```

Comment:
@piergm @javanna sharing for awareness, this test started to get flaky when consuming the new `include_empty_fields` API

This test is ingesting a new records with a new field, that sometimes wasn't returned on the first run of the field_caps API with `include_empty_fields=false`, there's now a retry implemented, making sure the test doesn't fail. Just wanted to ask that to make sure it works as expected:
When a new field was ingested with a new record, and the `field_caps` request with `include_empty_fields` set to false is requested to soon, the new field might not be returned with this request.

Reply:
Hi @kertal, this can be the case since the point-in-time view can be slightly outdate for field_caps (see this discussion in my pr https://github.com/elastic/elasticsearch/pull/103651#discussion_r1478207519). 
Instead of a retry I would suggest to call the PUT doc API with `?refresh=true`. This should make the test stable as was before

Reply:
This is because a refresh needs to happen. In my mind, this is better than before, because previously field caps could return new fields that search may not be able to see it (as it needs a refresh to make newly added documents searchable). Now, field caps with the new flag behaves similarly to search, in that it needs a refresh to see new fields that just received their first value. Now, there can always be additional refreshes between a field caps and the subsequent search call, so it does not mean that search does not see more than what field caps exposes, but it does not see less, if that makes sense.

Reply:
@javanna I agree that this is better, so far just this Kibana functional test had to be convinced that it is, I've applied @piergm's suggestion, and it works now with even less code. Thx both!

https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/5198

Reply:
@stratoula @davismcphee any objection moving forward, last change here, again stable, and less code is better code: https://github.com/elastic/kibana/pull/176885/commits/3b8be20dc6569cdbec6d6c5112ef7e234ddbed68

Reply:
No objections on my end!